src/WebJobs.Script.WebHost/Middleware/Buffering/ResponseBufferingMiddleware.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE_APACHE.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Buffering
+{
+    public class ResponseBufferingMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ResponseBufferingMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            var originalResponseBody = httpContext.Response.Body;
+
+            // no-op if buffering is already available.
+            if (originalResponseBody.CanSeek)
+            {
+                await _next(httpContext);
+                return;
+            }
+
+            var originalBufferingFeature = httpContext.Features.Get<IHttpBufferingFeature>();
+            try
+            {
+                // Shim the response stream
+                var bufferStream = new BufferingWriteStream(originalResponseBody);
```

Comment:
We should either do it with this PR or open another work item to track this, we should not allocate a BufferStream with the size of the respondeBody for non-proxy functions. this has a perf impact and I missed it when initially took a dependency on BufferingMiddleware code from aspnet team.

Reply:
I created an issue https://github.com/Azure/azure-functions-host/issues/3870